call-hierarchy-sample/src/model.ts

```diff
@@ -0,0 +1,91 @@
+import {
+	Range, Position, CallHierarchyProvider, TextDocument, CancellationToken, CallHierarchyItem,
+	SymbolKind, ProviderResult, CallHierarchyIncomingCall, CallHierarchyOutgoingCall, workspace, Uri
+} from 'vscode';
+
+/**
+ * Sample model of what the text in the document contains.
+ */
+export class FoodPyramid {
+	private relations: FoodRelation[] = [];
+	private nouns = new Set<string>();
+	private verbs = new Set<string>();
+
+	getRelationAt(wordRange: Range): FoodRelation | undefined {
+		return this.relations.find(relation => relation.range.contains(wordRange));
+	}
+
+	addRelation(relation: FoodRelation): void {
+		this.relations.push(relation);
+		this.nouns.add(relation.object).add(relation.subject);
+		this.verbs.add(relation.verb);
+	}
+
+	isVerb(name: string): boolean {
+		return this.verbs.has(name.toLowerCase());
+	}
+
+	isNoun(name: string): boolean {
+		return this.nouns.has(name.toLowerCase());
+	}
+
+	getVerbRelations(verb: string): FoodRelation[] {
+		return this.relations
+			.filter(relation => relation.verb === verb.toLowerCase());
+	}
+
+	getNounRelations(noun: string): FoodRelation[] {
+		return this.relations
+			.filter(relation => relation.involves(noun));
+	}
+
+	getSubjectRelations(subject: string): FoodRelation[] {
+		return this.relations
+			.filter(relation => relation.subject === subject.toLowerCase());
+	}
+
+	getObjectRelations(object: string): FoodRelation[] {
+		return this.relations
+			.filter(relation => relation.object === object.toLowerCase());
+	}
+}
+
+/**
+ * Model element.
+ */
+export class FoodRelation {
+	private _subject: string;
+	private _verb: string;
+	private _object: string;
```

Comment:
Align styles for privates with `FoodPyramid` and maybe others - my personal preference is `_xyz` ;-)

Reply:
I only used the `_xyz` names to avoid the clash with the attribute `get` methods. But granted and unified. Thanks. Done.